src/issues/github.rs

```diff
@@ -1,44 +1,79 @@
-use color_eyre::eyre::{eyre, Result};
+use color_eyre::eyre::{eyre, Result, WrapErr};
 
-use crate::app_config::get_or_prompt_for_octocrab;
+use crate::app_config::get_or_prompt_for_github_token;
 use crate::issues::Issue;
 use crate::{config, state};
 
-pub(crate) async fn list_issues(
+const ISSUES_QUERY: &str = r##"
+query($repo: String!, $owner: String!, $labels: [String!]) { 
+  repository(name:$repo, owner:$owner) { 
+    issues(states:OPEN, first: 30, labels: $labels) {
+      nodes {
+        number,
+        title
+      }
+    }
+  }
+}
+"##;
+
+#[derive(serde::Deserialize)]
+struct ResponseIssue {
+    number: usize,
+    title: String,
+}
+
+pub(crate) fn list_issues(
     github_config: Option<config::GitHub>,
     github_state: state::GitHub,
-    labels: Option<Vec<String>>,
+    labels: &Option<Vec<String>>,
 ) -> Result<(Option<config::GitHub>, state::GitHub, Vec<Issue>)> {
     match github_config {
         None => Err(eyre!("GitHub is not configured")),
```

Comment:
Technically outside the scope of the changes, but why is an `Option<config::GitHub>` accepted as a parameter when it's an error for it to be `None`?

Reply:
Probably a lazy refactor when breaking up the function above it 😂

Reply:
Fair enough. I'd suggest either fixing it here or making an issue for it to fix later.